src/Features/Core/Portable/DocumentationComments/CodeFixes/AbstractAddDocCommentParamNodeCodeFixProvider.cs

```diff
@@ -0,0 +1,161 @@
+        where TXmlElementSyntax : SyntaxNode
+        where TXmlNameAttributeSyntax : SyntaxNode
+        where TXmlTextSyntax : SyntaxNode
+        where TMemberDeclarationSyntax : SyntaxNode
+        where TParameterSyntax : SyntaxNode
+    {
+        public override FixAllProvider GetFixAllProvider() => WellKnownFixAllProviders.BatchFixer;
+
+        public async sealed override Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            var root = await context.Document.GetSyntaxRootAsync().ConfigureAwait(false);
+            var parameter = root.FindNode(context.Span) as TParameterSyntax;
+
+            var parentMethod = parameter?.FirstAncestorOrSelf<TMemberDeclarationSyntax>();
+            if (parentMethod != null)
+            {
+                context.RegisterCodeFix(
+                    new MyCodeAction(
+                        c => AddParamTagAsync(context.Document, context.Span, c), GetParameterName(parameter)),
+                    context.Diagnostics);
+            }
+        }
+        
+        protected abstract List<TXmlNameAttributeSyntax> GetNameAttributes(TXmlElementSyntax node);
+        protected abstract string GetValueFromNameAttribute(TXmlNameAttributeSyntax attribute);
+        protected abstract SyntaxNode GetDocCommentNode(SyntaxTriviaList parameter);
+        protected abstract string GetXmlElementLocalName(TXmlElementSyntax element);
+        protected abstract List<string> GetParameterNames(TMemberDeclarationSyntax method);
+        protected abstract string GetParameterName(TParameterSyntax parameter);
+        protected abstract TXmlElementSyntax GetNewNode(string parameterName, bool isFirstNodeInComment);
+
+        protected async Task<Document> AddParamTagAsync(
+            Document document, TextSpan span, CancellationToken cancellationToken)
+        {
+            var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var parentMethod = root.FindNode(span).FirstAncestorOrSelf<TMemberDeclarationSyntax>();
+            var docCommentNode = GetDocCommentNode(parentMethod.GetLeadingTrivia());
+
+            var newDocComment = docCommentNode;
+            var methodParamNames = GetParameterNames(parentMethod);
+
+            foreach (var name in methodParamNames)
+            {
+                var paramNodes = GetParamNodes(newDocComment);
+
+                if (NodeExists(paramNodes, name)) { continue; }
+
+                var paramsBeforeCurrentParam = methodParamNames.TakeWhile(t => t != name).ToList();
+                var paramsAfterCurrentParam = methodParamNames.Except(paramsBeforeCurrentParam).ToList();
+                paramsAfterCurrentParam.Remove(name);
+
+                SyntaxNode nodeBeforeNewParamNode = null;
+                SyntaxNode nodeAfterNewParamNode = null;
+                if (paramsBeforeCurrentParam.Any())
```

Comment:
Could you write a brief comment about what all this code is trying to accomplish?